src/subsystems/racket/discoverers/default/default.nix

```diff
@@ -0,0 +1,37 @@
+{
+  dlib,
+  lib,
+  subsystem,
+  ...
+}: let
+  l = lib // builtins;
+
+  discover = {
+    tree,
+    topLevel ? true,
+  }:
+    if (tree ? files."info.rkt")
+    then [
```

Comment:
According to this implementation, the recursion stops, whenever a `info.rkt` file is found.
If a racket project contains another racket project inside a sub-directory, the sub-project will be ignored. Should we cover that case as well?

Reply:
This was intentional on my part. The situation I wanted to avoid can be seen here:

- https://github.com/stchang/parsack/blob/master/parsack-lib/info.rkt
- https://github.com/stchang/parsack/blob/master/parsack-lib/parsack/info.rkt

The first of these two files is effectively the `package.json`. I have heard (don't remember where, but I want to say it was from the Racket Discord channel) that people sometimes use files like the second one, contained within top-level projects, to set variables, and not to declare sub-projects. So scanning indiscriminately for `info.rkt` files would incorrectly register the second `info.rkt` as a project root.

At the root of the aforementioned `parsack` repo, you will find that there are four sibling projects, each containing their own top-level `info.rkt`. This is the most common practice for repos that contain more than one package; I'm actually not aware of any scenarios where a sub-project is wholly contained within another project.